pkg/azdiskschedulerextender/azdiskschedulerextender.go

```diff
@@ -441,17 +434,93 @@ func setNodeScoresToZero(nodes []v1.Node) (priorityList schedulerapi.HostPriorit
 	return
 }
 
+func getRequestedVolumeSet(ns string, requestedVolumes []v1.Volume) (volumeSet map[string]struct{}, volumeCount int) {
+	// create a set of all volumes needed
+	volumeSet = map[string]struct{}{}
+	for _, volume := range requestedVolumes {
+		// not the same as len(pvNamesForRequestedVolumes). PV-s for a new pod might not be cached yet.
+		volumeCount++
+		if volume.PersistentVolumeClaim != nil {
+			if v, ok := pvcToPvMapCache.Load(getQualifiedName(ns, volume.PersistentVolumeClaim.ClaimName)); ok {
+				entry := v.(*pvcToPVEntry)
+				if !strings.EqualFold(entry.AccessMode, string(v1.ReadWriteMany)) &&
```

Comment:
I am not sure we need to check for RWO ROM on `Fileter` that's why this code was not refactored. For filtering we only want to subtract the volumes that are already attached to a node and not double count them in our maxcount assessment. So, if a RWM/ROM volume is attached to a node, so let be it, we don't want to ignore the attachment and risk incorrectly eliminating the node from our filtered list. It is important that we don't filter out too many nodes that can potentially host the pod because once we do, the node is not going to appear in rest of the scheduling.  On the other hand, for `Prioritize` we base our score on volumes that could potentially have replica attachments, thus checking for RWM and ROM makes sense. 

I think it is best to not refactor and reuse this function both for `Filter` and `Prioritize` because they have slightly different goals.

Reply:
I see your point. We can still refactor this portion by passing in `accessMode` filters as an argument. This is not necessarily a trivially small block of code and we should refactor when we can.